vertical-pod-autoscaler/enhancements/4016-in-place-updates-support/README.md

```diff
@@ -169,41 +208,19 @@ the same conditions that apply to updates in the `Recreate` mode.
 
 ### Test Plan
 
-The following test scenarios will be added to e2e tests. Both `InPlaceOnly` and `InPlaceOrRecreate` modes will be tested
-and they should behave the same:
+The following test scenarios will be added to e2e tests. The `InPlaceOrRecreate` mode will be
+tested in the following scenarios:
 
 * Admission controller applies recommendation to pod controlled by VPA. 
 * Disruption-free in-place update applied to all containers of a pod (request in recommendation bounds).
 * Partial disruption-free update applied to some containers of a pod, some disruptive changes skipped (request in
   recommendation bounds).
 * Disruptive in-place update applied to all containers of a pod (request out ouf recommendation bounds). 
-
-There will be also scenarios testing differences between `InPlaceOnly` and `InPlaceOrRecreate` modesL
-* Disruptive in-place update will fail. In `InPlaceOnly` pod should not be evicted, in the `InPlaceOrRecreate` pod
-  should be evicted and the recommendation applied.
-* VPA attempts an update that would change Pods QoS (`RequestsOnly` scaling, request initially < limit, recommendation
-  equal to limit). In `InPlaceOnly` pod should not be evicted, request slightly lower than the recommendation will be
-  applied. In the `InPlaceOrRecreate` pod should be evicted and the recommendation applied.
+* Disruptive in-place update will fail. Pod should be evicted and the recommendation applied.
+* Disruption free in-place updates fail, pod should not be evicted.
 
 ### Details still to consider
 
-#### Ensure in-place resize request doesn't cause restarts
-
-Currently the container [resize policy](https://kubernetes.io/docs/tasks/configure-pod-container/resize-container-resources/#container-resize-policies)
-can be either `NotRequired` or `RestartContainer`. With `NotRequired` in-place update could still end up
-restarting the container if in-place update is not possible, depending on kubelet and container
-runtime implementation. However in the proposed design it should be VPA's decision whether to fall back
-to restarts or not.
-
-Extending or changing the existing API for in-place updates is possible, e.g. adding a new
-`MustNotRestart` container resize policy.
-
-#### Should `InPlaceOnly` mode be dropped
-
-The use case for `InPlaceOnly` is not understood yet. Unless we have a strong signal it solves real
-needs we should not implement it. Also VPA cannot ensure no restart would happen unless
-*Ensure in-place resize request doesn't cause restarts* (see above) is solved.
-
 #### Careful with memory scale down
 
 Downsizing memory may have to be done slowly to prevent OOMs if application starts to allocate rapidly.
```

Comment:
Should we want to mention here that this will apparently always restart the container?

Reply:
This is indeed a KEP detail and I've added it, thanks!